Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -46,7 +46,10 @@ public async ValueTask<TItem> GetAsync(string id)
 
             TryLogDebugDetails(_logger, () => $"Read: {JsonConvert.SerializeObject(item)}");
 
-            return item;
+            if (item.Type == typeof(TItem).Name)
+                return item;
+
+            return null;
```

Comment:
```suggestion
            return string.IsNullOrEmpty(item.Type) || item.Type == typeof(TItem).Name
                ? item
                : null;
```

We need to account for the fact that some earlier versions of the SDK didn't have the `.Type` property. So when an `item` is hydrated, it's possible that the `.Type` property is `null`. This is why there is an `.IsDefined()` check on the expression based overload.